### ISSUE TITLE
incorrect discriminator loss in GAIA notebook

### ISSUE THREAD

**Issue by hanshengchiu** [timsainb/tensorflow2-generative-models#12]
The following is discriminator's loss in the GAIA notebook. Shouldn't the d_xg_loss be negative?
`disc_loss = d_xg_loss + d_x_loss - tf.clip_by_value(d_xi_loss, 0, d_x_loss)`

**Comment by Abel1802** [comment]
I encountered the same problem when I reproduced the code.

**Comment by Abel1802** [comment]
`gen_gradients = gen_tape.gradient(
            gen_loss, self.enc.trainable_variables + self.dec.trainable_variables
        ) 
`

**Comment by timsainb** [comment]
I should have mentioned that the notebook there was me experimenting with a variation of GAIA that I was trying out. I just added a line in the readme saying as much. But you're right. If someone wants to make a PR with the original algorithm for GAIA it would be much appreciated.

**Comment by Abel1802** [comment]
> I should have mentioned that the notebook there was me experimenting with a variation of GAIA that I was trying out. I just added a line in the readme saying as much. But you're right. If someone wants to make a PR with the original algorithm for GAIA it would be much appreciated.

Thank you!

**Comment by timsainb** [comment]
I've got a cleaner tensorflow 1 implementation of GAIA in a private repo that I used for the 2D meshplot examples figure from the paper  BTW. If anyone wants that code / access to that repo just ask. The repo is private just because its full of _Copy3.ipynb files with additional experiments that I never got around to making presentable.
